### ISSUE TITLE
Crash in 1.7.0 when compiled in release mode on new Telemetry

### ISSUE THREAD

**Issue by malcommac** [segmentio/analytics-swift#373]
**Describe the bug**
There is a serioous bug in Telemetry module when compiling the main app using release configuration with compiler optimization on. The entire app crash immediately at startup on iOS 18.x.

**To Reproduce**
1. Import segment 1.7.0
2. Compile app in release configuration
3. Launch and enjoy

**Expected behavior**
Not a crash for sure.

**Additional context**

Crash for thread:

```
Crashed: telemetryQueue
0  libdispatch.dylib              0x13ab0 __DISPATCH_WAIT_FOR_QUEUE__ + 484
1  libdispatch.dylib              0x135e8 _dispatch_sync_f_slow + 148
2  Indomio                        0x106e86c Telemetry.start() + 4386105452
3  Indomio                        0x1070238 implicit closure #2 in implicit closure #1 in Telemetry.subscribe(_:) + 4386112056
4  Indomio                        0x1070298 thunk for @escaping @callee_guaranteed (@guaranteed System) -> () + 4386112152
5  Indomio                        0x2d6154 partial apply for thunk for @escaping @callee_guaranteed (@in_guaranteed A._ValueReducer.Value) -> (@out ()) + 4371849556
6  Indomio                        0x2d6154 partial apply for thunk for @escaping @callee_guaranteed (@in_guaranteed A._ValueReducer.Value) -> (@out ()) + 4371849556
7  Indomio                        0x1081160 partial apply for closure #1 in Store.notify<A>(subscribers:state:) + 4386181472
8  Indomio                        0xee4e8 thunk for @escaping @callee_guaranteed @Sendable () -> () + 4369851624
9  libdispatch.dylib              0x2370 _dispatch_call_block_and_release + 32
10 libdispatch.dylib              0x40d0 _dispatch_client_callout + 20
11 libdispatch.dylib              0xb6d8 _dispatch_lane_serial_drain + 744
12 libdispatch.dylib              0xc1e0 _dispatch_lane_invoke + 380
13 libdispatch.dylib              0x17258 _dispatch_root_queue_drain_deferred_wlh + 288
14 libdispatch.dylib              0x16aa4 _dispatch_workloop_worker_thread + 540
15 libsystem_pthread.dylib        0x4c7c _pthread_wqthread + 288
16 libsystem_pthread.dylib        0x1488 start_wqthread + 8
```

**Comment by dmcgloin** [comment]
There's a fix in 1.7.1. Worked for us.

**Comment by bsneed** [comment]
Thanks @dmcgloin; @malcommac please file a new ticket if you have any issues with 1.7.1.
